git-branchless/src/commands/move.rs

```diff
@@ -168,16 +217,163 @@ pub fn r#move(
+                // components, then it's possible that the current component
+                // could have multiple possible parents.
+                //
+                // To check for this, we can confirm that the nearest
+                // destination component is an ancestor of the previous (ie next
+                // nearest). This works because possible_destinations is made
+                // from component_roots, which has been sorted topologically; so
+                // each included component should "come after" the previous
+                // component.
+                let nearest_dest_index = possible_destinations.len() - 1;
+                if possible_destinations.len() > 1
+                    && !dag.query().is_ancestor(
+                        possible_destinations[nearest_dest_index - 1].into(),
+                        possible_destinations[nearest_dest_index].into(),
+                    )?
+                {
+                    writeln!(
+                        effects.get_output_stream(),
+                        "This operation cannot be completed because the {} at {}\n\
+                        has multiple possible parents also being moved. Please retry this operation\n\
+                        without this {}, or with only 1 possible parent.",
+                        if component.count()? == 1 {
+                            "commit"
+                        } else {
+                            "range of commits rooted"
+                        },
+                        component_root,
+                        if component.count()? == 1 {
+                            "commit"
+                        } else {
+                            "range of commits"
+                        },
+                    )?;
+                    return Ok(ExitCode(1));
+                }
+
+                let nearest_component =
+                    exact_components.get(&possible_destinations[nearest_dest_index]);
+                if let Some(nearest_component) = nearest_component {
+                    // The current component could be descended from any commit
+                    // in nearest_component, not just it's head.
+                    let dest_ancestor = dag
+                        .query()
+                        .ancestors(CommitSet::from(component_head))?
+                        .intersection(nearest_component);
+                    let head = dag.query().heads(dest_ancestor.clone())?.first()?;
+                    if let Some(head) = head {
+                        NonZeroOid::try_from(head)?
+                    } else {
+                        dest_oid
+                    }
+                } else {
+                    // TODO this should be unreachable
+                    dest_oid
+                }
+            };
+
+            let component_parent = match dag.get_only_parent_oid(*component_root) {
+                Ok(oid) => oid,
+                Err(_) => {
```

Comment:
In this case, `dag.get_only_parent_oid` could have had an I/O error, which we would want to propagate directly. Instead, either make `get_only_parent_oid` return an `eyre::Result<Option<_>>`, or get all of the parents and do a pattern match.

Reply:
OK, done, thank you. 

Do you have tips on where to look to know if something will fail due to I/O vs just failing for normal use situations?